python/src/y_sweet_sdk/update.py

```diff
@@ -0,0 +1,46 @@
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    # Avoid circular import
+    from .connection import DocConnection
```

Comment:
@greptileai is this python best practice, or should we avoid conditional imports? What would you suggest to avoid the circular import we're concerned about here?

Reply:
Cyclical imports _of types_ are generally impossible to avoid when using type annotations. Unfortunately, unlike modern JavaScript, Python doesn't have an `import type` syntax, it has this ugly monstrosity instead.

SO ref: https://stackoverflow.com/a/39757388

IMHO the only real alternative here is not using python's half-baked type annotations, but I still think we do better by having them.